packages/shared/src/server/repositories/traces.ts

```diff
@@ -0,0 +1,159 @@
+import { TraceClickhouseRecord } from "../clickhouse/schema";
+import { queryClickhouse } from "./clickhouse";
+import {
+  createFilterFromFilterState,
+  getProjectIdDefaultFilter,
+} from "../queries/clickhouse-filter/factory";
+import { ObservationLevel } from "@prisma/client";
+import { FilterState } from "../../types";
+import { logger } from "../logger";
+
+export type TracesTableReturnType = Pick<
+  TraceClickhouseRecord,
+  | "project_id"
+  | "id"
+  | "name"
+  | "timestamp"
+  | "bookmarked"
+  | "release"
+  | "version"
+  | "user_id"
+  | "session_id"
+  | "tags"
+  | "metadata"
+  | "public"
+> & {
+  level: ObservationLevel;
+  observation_count: number | null;
+  latency: number;
+  usage_details: Record<string, number>;
+  cost_details: Record<string, number>;
+  scores_avg: Array<{ name: string; avg_value: number }>;
+};
+
+export const getTracesTable = async (
```

Comment:
Overall, I think this makes for a good DX until we find a better solution or an ORM. As long as we keep all within a single file that should be good.